### ISSUE TITLE
Code slower at beginning of program

### ISSUE THREAD

**Issue by LoopyAshy** [beefytech/Beef#476]
I am confused to be honest: I was messing around out of boredom and it occurred to me that the first loop which calls console.writeline in the program I was playing with always seemed drastically slower, I had tried setting several settings to see if that would fix it and alas no, this isn't exactly the end of the world but I am genuinely curious to why this is happening.
I maybe completely wrong here but it seems like its nothing to do with the loops themselves and its simply some kind of warm up lag, I am not sure what could be causing that and even added it to my anti virus exclusions just in case. most confusing part about this however is: Thread.sleep makes no difference even if slept for same time the loop would of taken, however if done via loops: every 1000 loops takes off a chunk of time for the next loop which is noticeably linear.

I imagine this could be something obvious and I am making a fool of myself, but I am genuinely curious to why this is happening.

### Examples
1. A simple for loop which starts at 0 and ends at 10000 and simply prints to console, takes 3111ms on first for loop then 780ms on all for loops after.
2. A simple for loop which iterates through a list of Booleans and simply prints to console, takes 3466ms on first for loop then 754ms on all for loops after.
3. same as 1 however it sleeps for 4 seconds prior, takes same time as 1
4. same as 1 however it does 1000 loops which prints to console before the first 10000 for loop, the for loop took 2941ms.
5, same as 4 however it does 5000 loops which prints to console before the first 10000 for loop, the for loop took 1826ms.

## Notes
1. I have only tested with console printing (lazy I know), however when doing none-console print loops it usually elapsed too quickly for observation.
2. Amount of code before the loop doesn't seem to make a difference.

**Comment by bfiete** [comment]
This seems like a case where posting an example program would have been a lot easier than describing the issue.

**Comment by bfiete** [comment]
Okay- so I built a test program like you described. I added magic feature, however: I invoke the debugger's profiler to capture each iteration through the loop into separate profile runs.
```C#
public static int Main(String[] args)
{
	for (int j < 5)
	{
		Stopwatch sw = scope .();
		sw.Start();
		let instance = Profiler.StartSampling().Value;
		for (int i < 5000)
			Console.WriteLine("{}", i);
		sw.Stop();
		instance.Dispose();
		Debug.WriteLine("{}", sw.ElapsedMilliseconds);
	}
	return 0;
}
```

On my machine the first iteration takes 1571ms with 1520ms spent in NtWriteFile and 2% spent in ZwFlushBuffersFile, and the last iteration takes 443ms with 70% spent in NtWriteFile and 24% spent in ZwFlushBuffersFile. Why is that the case? I don't know,   but it's something internal to Windows. The console isn't meant to be a high-throughput output device.
